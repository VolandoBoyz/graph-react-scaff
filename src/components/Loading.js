import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Loading = props =>
  <div className={props.className}>
    <span>ðŸ˜ƒ</span>
  </div>;

const LoadingStyled = styled(Loading)`
    display: flex;
    height: 100%;
    justify-content: center;
    align-items: center;
    span {
      animation-name: spin;
      animation-duration: 400ms;
      animation-iteration-count: infinite;
      animation-timing-function: linear;
      font-size: 50px;
    }

`;

Loading.propTypes = {
  className: PropTypes.string.isRequired,
};


export default LoadingStyled;
