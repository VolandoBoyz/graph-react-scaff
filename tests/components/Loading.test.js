import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../../src/components/Loading';

test('Loading renders as expected', () => {
  const component = renderer.create(
    <Loading />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
