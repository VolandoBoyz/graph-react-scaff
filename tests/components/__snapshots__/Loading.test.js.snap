// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders as expected 1`] = `
<div
  className="iDojEz"
>
  <span>
    ðŸ˜ƒ
  </span>
</div>
`;
